src/Common/TraceCollector.cpp

```diff
@@ -56,6 +58,15 @@ TraceCollector::~TraceCollector()
 
 void TraceCollector::collect(TraceType trace_type, const StackTrace & stack_trace, Int64 size)
 {
+    /// It is possible that the method will be called recursively
+    /// (one from MemoryTracker and another via signal handler).
+    ///
+    /// Better to drop these events to avoid trace events overlaps.
```

Comment:
But what's wrong with event overlaps?
I think it's normal...

Reply:
The problem here is that `buf_size` = 5915 (due to `sizeof(StackTrace::Frames)`, which size is 4864), while atomically you can write only `PIPE_BUF` into pipe, which is `4096`
(In debug build it will be `buf_size` = 10779 and `sizeof(StackTrace::Frames)` = 9728)

Reply:
We should only pass FramePointers, why Frames?

Reply:
This is obviously a bug, introduced here: #11300

Reply:
>We should only pass FramePointers, why Frames?

Indeed, and only frame pointers are written, only the calculation of buf_size wrong.